src/middlewares/validateRejectTrip.js

```diff
@@ -0,0 +1,85 @@
+import Joi from 'joi';
+import Models from 'database/models';
+import { errorResponse } from 'utils/responses';
+import setLanguage from '../utils/international';
+
+const { travelRequests } = Models;
+
+export const isUuidParamValid = (req, res, next) => {
+  const {
+    user: {
+      preferedLang
+    }
+  } = req;
+  const uuiDchemas = Joi.object().keys({
+    tripId: Joi.string().guid(),
+  });
+  const schema = Joi.validate(req.params, uuiDchemas);
+  if (schema.error) {
+    errorResponse(res, 400, setLanguage(preferedLang).__('PleaseUseValidId'));
+  }
+  next();
+};
+
+export const isTripExist = async (req, res, next) => {
+  const {
+    user: {
+      preferedLang
+    },
+    params: {
+      tripId
+    }
+  } = req;
+  const trip = await travelRequests.findOne({ where: { requestId: tripId } });
+  if (!trip) {
+    errorResponse(res, 404, setLanguage(preferedLang).__('TripNotFound'));
+  }
+  next();
+};
+export const isTripAssigned = async (req, res, next) => {
+  const {
+      user: {
+        email,
+        preferedLang
+      },
+      params: {
+        tripId
+      }
+    } = req,
+    foundAssign = await travelRequests.findOne({ where: { manager: email, requestId: tripId } });
+  if (!foundAssign) {
+    errorResponse(res, 403, setLanguage(preferedLang).__('TripNotAssigned'));
+  }
+  next();
+};
+export const isTripRejected = async (req, res, next) => {
+  const {
+      params: {
+        tripId
+      },
+      user: {
+        preferedLang
+      }
+    } = req,
+    rejectedTrip = await travelRequests.findOne({ where: { requestId: tripId } });
+  if (rejectedTrip.status === 'rejected') {
+    errorResponse(res, 403, setLanguage(preferedLang).__('TripAlreadyRejected'));
+  }
+  next();
+};
+
+export const isTripApproved = async (req, res, next) => {
+  const {
+      params: {
+        tripId
+      },
+      user: {
+        preferedLang
+      }
+    } = req,
+    approvedTrip = await travelRequests.findOne({ where: { requestId: tripId } });
+  if (approvedTrip.status === 'approved') {
+    errorResponse(res, 403, setLanguage(preferedLang).__('NotAllowedToRejecApprovedRequest'));
+  }
+  next();
```

Comment:
@hakoemmy looking good, all validations handled 
but there is a way that you would have done all of this with just one query or just two

Reply:
@blaise82  Thank you very much for such feedback. Let me endeavor to optimize this in a short way.

Reply:
@blaise82 I made some refactorings. Thank you again for the feedback